### PR TITLE
xpaste: init at 1.5

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

[Xpaste](https://github.com/ossobv/xpaste) allows pasting text by emulating keypresses. Useful for KVM windows.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/xpaste/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+python3Packages.buildPythonApplication rec {
+  pname = "xpaste";
+  version = "1.5";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "ossobv";
+    repo = pname;
+    rev = "refs/tags/v${version}";
```
Comment (azahi) [inline pkgs/tools/text/xpaste/default.nix:14]:
```suggestion
    rev = "v${version}";
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/tools/text/xpaste/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/text/xpaste/default.nix:14]: fine either way

File: pkgs/tools/text/xpaste/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    hash = "sha256-Z9YxhVP+FUvWET91ob4SPSW+XX7/wzlgPr53vO517L4=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    xlib
+  ];
+
+  # no tests
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/text/xpaste/default.nix:23]: Can we do an pythonImportsCheck instead?

Comment (gador) [reply pkgs/tools/text/xpaste/default.nix:23]: No, upstream does not provide a module which could be imported.
`site-pakckages` does not contain a module in `out/lib/python3.10/site-packages`:

```
ls -la result/lib/python3.10/site-packages
dr-xr-xr-x - root  1 Jan  1970 xpaste-1.5.dist-info
```

**Comment by gador** [comment]
discovered an issue with x11 pasting. Added a commit to fetch the upstream patch.
@SuperSandro2000 added a comment about the checks (which do not exist..)

File: pkgs/tools/text/xpaste/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  ];
+
+  patches = [
+  (fetchpatch {
+    # https://github.com/ossobv/xpaste/pull/6
+    name = "fix-function-call-after-wayland-update.patch";
+    url = "https://github.com/ossobv/xpaste/commit/47412738dad4b5fc8bc287ead23c8440bfdc547d.patch";
+    hash = "sha256-t4LZG600AsFTtKjXCxioGcAP4YcHIdQ/fVMIYjsunuA=";
+  })
+];
```
Comment (SuperSandro2000) [inline pkgs/tools/text/xpaste/default.nix:30]:
```suggestion
    (fetchpatch {
      # https://github.com/ossobv/xpaste/pull/6
      name = "fix-function-call-after-wayland-update.patch";
      url = "https://github.com/ossobv/xpaste/commit/47412738dad4b5fc8bc287ead23c8440bfdc547d.patch";
      hash = "sha256-t4LZG600AsFTtKjXCxioGcAP4YcHIdQ/fVMIYjsunuA=";
    })
  ];
```

**Pull request merged**